extensions/amp-story-page-attachment/0.1/amp-story-page-attachment.js

```diff
@@ -394,10 +400,89 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
     );
   }
 
+  /**
+   * Get outlink element.
+   * @return {?Element}
+   * @private
+   */
+  getOutlinkEl_() {
+    if (!this.outlinkEl_) {
+      this.outlinkEl_ = this.element.parentElement
+        .querySelector('amp-story-page-outlink')
+        ?.querySelector('a');
+    }
+    return this.outlinkEl_;
+  }
+
+  /**
+   * Get legacy outlink element.
+   * @return {?Element}
+   * @private
+   */
+  getLegacyOutlinkEl_() {
+    if (!this.legacyOutlinkEl_) {
+      this.legacyOutlinkEl_ = this.element.parentElement
+        .querySelector('.i-amphtml-story-page-open-attachment-host')
+        ?.shadowRoot?.querySelector('a.i-amphtml-story-page-open-attachment');
+    }
+    return this.legacyOutlinkEl_;
+  }
+
+  /**
+   * Check if two URLs have the same origin and path but different hashes at the end.
+   * @param {string} url1
+   * @param {string} url2
+   * @return {boolean}
+   * @private
+   */
+  urlsHaveSameOriginAndPath_(url1, url2) {
+    if (!url1 || !url2) {
+      return false;
+    }
+
+    const url1WithoutHash = url1.split('#')[0];
+    const url2WithoutHash = url2.split('#')[0];
+
+    return url1WithoutHash === url2WithoutHash;
+  }
+
+  /**
+   * If the element is an amp-story-page-outlink the click target is its anchor element child.
+   * This is for SEO and analytics optimisation.
+   * Otherwise the element is the legacy version, amp-story-page-attachment with an href,
+   * and a click target is the button built by the component.
+   * @private
+   */
+  programmaticallyClickOnOutlink_() {
+    const pageOutlinkChild = this.getOutlinkEl_();
+    const pageAttachmentChild = this.getLegacyOutlinkEl_();
+    if (pageOutlinkChild) {
+      pageOutlinkChild.click();
+    } else if (pageAttachmentChild) {
+      triggerClickFromLightDom(pageAttachmentChild, this.element);
+    }
+  }
+
   /**
    * @override
    */
   open(shouldAnimate = true) {
+    // If the target is a branching link, redirect immediately without opening the drawer.
+    const outlinkEl = this.getOutlinkEl_() || this.getLegacyOutlinkEl_();
+    if (
+      this.urlsHaveSameOriginAndPath_(window.location.href, outlinkEl?.href) &&
+      outlinkEl?.href.includes('#page=')
```

Comment:
qq, what does the `#page=` hash do/identify?

Reply:
The link with this is a branching link that links directly to the specific page with the provided page id. You can find more details here: https://amp.dev/documentation/components/amp-story#branching